### PR TITLE
Add Signature Counts to Console

### PR DESCRIPTION
**Pull request by quantumagi**
See #316

### CODE REVIEW

File: src/Stratis.FederatedPeg.Features.FederationGateway/TargetChain/CrossChainTransfer.cs [hunk 1]
```diff
@@ -194,6 +194,21 @@ public static bool TemplatesMatch(Network network, Transaction partialTransactio
...
+            Guard.NotNull(this.PartialTransaction, nameof(this.PartialTransaction));
+            Guard.Assert(this.PartialTransaction.Inputs.Any());
+
+            Script scriptSig = this.PartialTransaction.Inputs[0].ScriptSig;
+            if (scriptSig == null)
+                return 0;
+
+            Op[] ops = scriptSig.ToOps().ToArray();
+
+            return ops.Length - (1 + ops.Count(o => o.Code == OpcodeType.OP_0));
```
Comment (zeptin) [inline src/Stratis.FederatedPeg.Features.FederationGateway/TargetChain/CrossChainTransfer.cs]: This seems a bit hacky, doesn't the P2SH template have a helper method for extracting the parameters? Or does that not work with a partially signed transaction?

Comment (bokobza) [reply src/Stratis.FederatedPeg.Features.FederationGateway/TargetChain/CrossChainTransfer.cs]: can use
`TransactionSignature[] result = PayToMultiSigTemplate.Instance.ExtractScriptSigParameters(this.network, new Script(scriptSig));`

**Comment by monsieurleberre** [comment]
I have put your suggestion in Jeremy, let me know what you think

**Comment by bokobza** [comment]
looks great

**Pull request merged**